src/intel_driver.h

```diff
@@ -111,6 +111,18 @@ extern uint32_t g_intel_debug_option_flags;
 #define VA_INTEL_DEBUG_OPTION_BENCH     (1 << 1)
 #define VA_INTEL_DEBUG_OPTION_DUMP_AUB  (1 << 2)
 
+#define LOCAL_I915_CONTEXT_PARAM_WATCHDOG 0x7
+#define LOCAL_I915_EXEC_VEBOX             (4<<0)
+#define LOCAL_I915_EXEC_VCS2              (5<<0)
+
+typedef enum _gen_gpu_class_engine {
+    RENDER_CLASS = 0,
+    VIDEO_DECODE_CLASS,
+    VIDEO_ENHANCEMENT_CLASS,
+    COPY_ENGINE_CLASS,
+    GEN_GPU_ENGINE_NUMBER
+} _gen_gpu_class_engine, *pgen_gpu_class_engine;
+
```

Comment:
Above looks a copy of enum drm_i915_gem_engine_class from i915_drm.h, apart from the last entry which I don't understand.

Reply:
I took it from an internal patch from a different implementation for the iHD media driver thinking that was just a simple list of enums of the different type of engines however, using the struct from i915_drm.h after bumping libdrm to version 4.92 that includes this:

/*
 * Different engines serve different roles, and there may be more than one
 * engine serving each role. enum drm_i915_gem_engine_class provides a
 * classification of the role of the engine, which may be used when requesting
 * operations to be performed on a certain subset of engines, or for providing
 * information about that group.
 */
enum drm_i915_gem_engine_class {
	I915_ENGINE_CLASS_RENDER	= 0,
	I915_ENGINE_CLASS_COPY		= 1,
	I915_ENGINE_CLASS_VIDEO		= 2,
	I915_ENGINE_CLASS_VIDEO_ENHANCE	= 3,

	I915_ENGINE_CLASS_INVALID	= -1
};

I end up with a GPU hang. Note that the struct are similars but not identical, specifically, VIDEO_DECODE_CLASS is 1 on my definition whereas I915_ENGINE_CLASS_VIDEO is 2 in i915_drm.h.